core/src/main/java/net/adoptopenjdk/icedteaweb/client/controlpanel/ControlPanel.java

```diff
@@ -206,7 +206,7 @@ public void actionPerformed(ActionEvent e) {
                             Translator.R("CPJVMconfirmReset"),
                             Translator.R("CPJVMconfirmReset"), JOptionPane.OK_CANCEL_OPTION);
                     if (i == JOptionPane.OK_OPTION) {
-                        jvmPanel.resetTestFieldArgumentsExec();
+                        //jvmPanel.resetTestFieldArgumentsExec();
```

Comment:
the OK button has no longer any functionality?

Reply:
oh, missed that

Reply:
I think we should discuss this code block. When pressing "OK" everything without this code block will be executed. This code block will only be executed when pressing "APPLY". @judovana I do not understand this specific case for 100%. Can you give me a hint?

Reply:
Hi! Was not rechecking the code, but the userexperiecne should be:
cancel - close without saving
apply - save, but keep window opened
ok - apply and close window

Reply:
@judovana
Agree on the user experience. Because I am pretty new to swing (manly did backend stuff) I tried to understand how the saving works exactly. As far as I read the code the following happens:
- whenever a value is entered or modified by the user the class `DocumentAdapter` propagates the changes immediately to in memory copy of the settings.
- when the user hits apply the in memory copy of the settings are written back to the file system.
- after the config has been persisted the JVM setting is validated and if found invalid a dialog with the option to `reset` is displayed. If the users chooses `OK` then the JVM setting is cleared (blank string)

But the invalid JVM setting has already been persisted to the file system and clearing the entry in the in memory copy of the config will only be persisted to the file system if the user closes the control panel with either the OK or the Apply button.

Did I understand the code correctly?
Do you agree with me that the current behavior is not exactly what the user would expect?

I would suggest to validate the content of a pane before the config is written to the file system

Reply:
I opened a new issue (#272 ) for this and will merge this branch for now.